library/std/src/sys/unix/stack_overflow.rs

```diff
@@ -54,6 +54,22 @@ mod imp {
     use crate::sys::unix::os::page_size;
     use crate::sys_common::thread_info;
 
+    #[cfg(any(target_os = "linux", target_os = "android"))]
```

Comment:
Looking at the linux-like targets in libc, it looks like only uclibc is missing the `si_addr()` accessor at the moment.  Would it make sense to leave the existing logic in place, adding the supplemental accessor definition for only `target_env = uclibc`?